### ISSUE TITLE
code is automatically deleted and crashed when using shortcut in eclipse + vrapper environment

### ISSUE THREAD

**Issue by jjunest** [vrapper/vrapper#794]
i'm developing springs using eclipse with vrapper plugins to use shorcut keys.

but sometimes when using shotcuts like ctrl+z or ctrl+c (i don't know the exactly when the problems happens... hard to detect the timing) ..

some codes are deleted and crashed automatically, so all code i worked became trash...\

the more suprise is that crahsed parts were not where i was coding. 

the crashed and deleted parts were far away from where i was editing the codes.

left picture is crashed code parts and right picture is original template code which i worked on.

please help me....

what's the problem? any clues? thanks in advance :)

```
				<div class="categoryDetails">
								<ul class="list-inline rating">
									<li><i class="fa fa-star" aria-hidden="true"></i></li>
									<li><i class="fa fa-star" aria-hidden="true"></i></li>
									<li><i class="fa fa-star" aria-hidden="true"></i></li>
									<li><i class="fa fa-star" aria-hidden="true"></i></li>
									<li><i class="fa fa-star" aria-hidden="true"></i></li>
								</ul>
								<h2><a href="blog-details.html" style="color: #222222">Glory Hole Doughnuts</a> <span class="likeCount"><i class="fa fa-heart-o" aria-hidden="true"></i> 10 k</span></h2>
								<p>1569 Queen Street West <span class="placeName">Toronto</span></p>
								<p>Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed eiusmod tempor incididunt  labore et dolore magna aliqua. </p>
								<ul class="list-inline list-tag">
									<li><a href="category-list-full.html">Hotel,</a></li>
									<li><a href="category-list-full.html">Restaurant</a></li>
								</ul>
							</div>
```

----------------------------------------------------------------------------------------------
```
<div class="categoryDetails">
<ul class="list-inline rating">
<li><i class="fa fa-star" aria-hidden="true"></i></li>
<li><i class="fa fa-star" aria-hidden="true"></i></li>
<li><i class="fa fa-star" aria-hidden="true"></i></li>
<li><i class="fa fa-star" aria-hidden="true"></i></li>
<li><i class="fa fa-star" aria-hidden="true"></i></li>
</ul>
<h2>
<a href="blog-details.html" style="color: #222222">${listValue.hmcNm}</a>
<span class="likeCount"><i class="fa fa-heart-o"	aria-hidden="true"></i> 10 k</span>
</h2>

<p>
aceName">${listValuemNo}</span>
</p>
<p>Lorem ips olor sit amet, consectetur adipi ng elit, susmod tempor incididunt labore et dolore mag aliqua.</p>lass="lis ine list-tag"> ><a href="category-list-fu
				ml">Hotel,</a>
											</li>

											<li><a href="category-li
				ll.html">Restaurant</a>
											</l
												</ul>
										</div>
```

Screenshot:
https://i.stack.imgur.com/iIdjZ.png

**Comment by albertdev** [comment]
I'm sorry to hear that your code got corrupted.

Sadly, I have never seen anything like it during recent years. Did undo still work if you press <kbd>Ctrl+Z</kbd> so that your code reverted to the previous non-broken source?

What version of Eclipse and Vrapper are you running? Likely Eclipse Oxygen and Vrapper 0.72.0, but I'd like to be sure.

What kind of sources where you editing, and what kind of Eclipse plugins do you have installed to edit them? There might be differences between the "Eclipse IDE for Java EE Developers" (which I'm using) and e.g. "Eclipse for JavaScript and Web Developers".

**Comment by albertdev** [comment]
I'm not entirely sure I can remove the Unconfirmed label, though I did appear to see a corrupted editor after doing extract method, switching to another editor to yank something, pasting it in the original editor and then triggering Undo a few times (though the `u` key bind).

The code in the editor would be nearly all gone, but the weirdest of all was that the line numbers would be all over the place, making it look like a visual problem. I believe it went away when I ran `:edit!` (thus dropping all changes).
